### PR TITLE
28 - Create npm module for the design system

### LINKED ISSUE #28: Create a npm module for the Dataverse Design System so the community can use it for their applications and future plugins
## Overview of the Feature Request

The goal is to create a npm module that includes the elements of the Dataverse Design System, decoupled from the frontend application. The initial elements to be included in this plugin will be identified in #14.

This npm module will have an associated GitHub repository, as well as independent versioning. The module will be installed on the Dataverse Frontend application, as well as any other application of the community which needs to use the Dataverse UI elements. The future Dataverse plugins, as described in #23, will be able to use this module as well.

We will develop the UI elements identified in #14 directly on this module.

## What kind of user is the feature intended for?
Dataverse frontend developers

## What inspired the request?
- Frontend subgroup weekly meeting
- #23 

## What existing behavior do you want changed?
N/A

## Any brand new behavior do you want to add to Dataverse Frontend?
N/A

## Any related open or closed issues to this feature request?
- #14 
- #23 
- #39

### PR DESCRIPTION
**Pull request by MellyGray**
## What this PR does / why we need it:
This PR creates an npm module that includes the elements of the Dataverse Design System, decoupled from the frontend application. 

The intention behind this PR is that we have an npm module with all the Design System elements so that anyone can use them in their own projects

## Which issue(s) this PR closes:

- Closes #28 

## Special notes for your reviewer:
This new library has been called dataverse-design-system and can be used like `import { Navbar } from 'dataverse-design-system' `

We can develop the page sections while updating the dataverse-design-system at the same time because we have everything in one repository.

## Suggestions on how to test this:

1. Build the module
`npm run install`

`cd packages/design-system`

`npm run build`

2. Use the module in the main application

Go back to the root `dataverse-frontend/`

`npm run start`

The application runs normally using the dataverse-design-system components

`npm run storybook`

This command will open both the storybook of the app and the design system to facilitate the development workflow.

3. Run tests

To run the tests we can run the commands in the root of /dataverse-frontend and it will run dataverse-design-system tests too

`npm run test:unit-all`

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
None

## Is there a release notes update needed for this change?:
dataverse-design-system module created to have a Design System module that could later on be published in npm. This way anyone could use the Dataverse Design System in their projects

### CODE REVIEW

**Review by ekraffmiller — COMMENTED** [review]
Looks good!  I know this branch is behind develop, I just wanted to check - when this branch merges with develop, are we going to update the pre-commit check and test.yml that it does cypress component testing and storybook accessibility testing for both the design system and for app-specific components?

File: packages/design-system/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "dataverse-design-system",
```
Comment (pdurbin) [inline packages/design-system/package.json:2]: The pull request description says, "This new library has been called dataverse-ui-lib."

It should be updated, right?

**Review by pdurbin — COMMENTED** [review]
Overall, this is looking good! I was able to build the new package and the frontend with it.

I left some comments. Merge conflicts need to be resolved, of course.

**Comment by MellyGray** [comment]
> > `cd packages/dataverse-ui-lib`
> 
> Should this be `cd packages/design-system`?

Sure! I forgot to update the description. Done!

**Comment by pdurbin** [comment]
@MellyGray thanks!

Just a sanity check... this PR is not targeting "develop". Is that on purpose?

**Comment by MellyGray** [comment]
> @MellyGray thanks!
> 
> Just a sanity check... this PR is not targeting "develop". Is that on purpose?

Yes, it's on purpose. I wanted to fix the merge conflicts pointing to the cypress-coverage instead of pointing to develop, because there was a lot of work to fix those conflicts, I even had to add more tests, move folders around, etc. So I decided to fix that as soon as possible

So yes, the intention is to merge first the cypress-coverage branch and then this branch.

**Review by ekraffmiller — APPROVED** [review]
LGTM

**Comment by kcondon** [comment]
@MellyGray All tests pass, thanks. I did see an error while building. Is this a concern?

vite:dts] Start generate declaration files...
src/lib/components/theme/ThemeContext.ts:4:14 - error TS4023: Exported variable 'ThemeContext' has or is using name 'BaseTheme' from external module "/Users/kcondon/testing/qa/spa/dataverse-frontend/packages/design-system/src/lib/components/theme/BaseTheme" but cannot be named.

4 export const ThemeContext = createContext(baseTheme)
               ~~~~~~~~~~~~
src/lib/components/theme/ThemeProvider.tsx:15:14 - error TS4023: Exported variable 'useTheme' has or is using name 'BaseTheme' from external module "/Users/kcondon/testing/qa/spa/dataverse-frontend/packages/design-system/src/lib/components/theme/BaseTheme" but cannot be named.

15 export const useTheme = () => useContext(ThemeContext)
                ~~~~~~~~

[vite:dts] Declaration files built in 4668ms.

**Comment by MellyGray** [comment]
> @MellyGray All tests pass, thanks. I did see an error while building. Is this a concern?
> 

It's a concern. I fixed it, thanks!

I also went through some merge conflicts. I had some problems with the Gihub Actions but now it seems fine

**Pull request merged**